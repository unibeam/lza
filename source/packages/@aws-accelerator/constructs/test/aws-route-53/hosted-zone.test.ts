/**
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */

import * as cdk from 'aws-cdk-lib';
import { HostedZone } from '../../index';
import { snapShotTest } from '../snapshot-test';

const testNamePrefix = 'Construct(HostedZone): ';

//Initialize stack for snapshot test and resource configuration test
const stack = new cdk.Stack();
const hostedZoneName = HostedZone.getHostedZoneNameForService('s3-global.accesspoint', stack.region);
const ecrApiHostedZoneName = HostedZone.getHostedZoneNameForService('ecr.api', stack.region);
const appstreamApiHostedZoneName = HostedZone.getHostedZoneNameForService('appstream.api', stack.region);
const deviceAdvisorIotHostedZoneName = HostedZone.getHostedZoneNameForService('deviceadvisor.iot', stack.region);
const pintpointSmsVoiceV2HostedZoneName = HostedZone.getHostedZoneNameForService('pinpoint-sms-voice-v2', stack.region);
const rumDataplaneHostedZoneName = HostedZone.getHostedZoneNameForService('rum-dataplane', stack.region);
const ecsAgentHostedZoneName = HostedZone.getHostedZoneNameForService('ecs-agent', stack.region);
const ecsTelemetryHostedZoneName = HostedZone.getHostedZoneNameForService('ecs-telemetry', stack.region);
const sageMakerNotebookHostedZoneName = HostedZone.getHostedZoneNameForService('notebook', stack.region);
const sageMakerStudioHostedZoneName = HostedZone.getHostedZoneNameForService('studio', stack.region);
const s3HostedHostedZoneName = HostedZone.getHostedZoneNameForService('s3', stack.region);
const codeArtifactRepositoriesHostedZoneName = HostedZone.getHostedZoneNameForService(
  'codeartifact.repositories',
  stack.region,
);
const codeArtifactApiHostedZoneName = HostedZone.getHostedZoneNameForService('codeartifact.api', stack.region);

new HostedZone(stack, `TestHostedZone`, {
  hostedZoneName,
  vpcId: 'Test',
});

new HostedZone(stack, `codeArtifactApiHostedZone`, {
  hostedZoneName: codeArtifactApiHostedZoneName,
  vpcId: 'Test',
  tags: [{ key: 'key', value: 'value' }],
});

new HostedZone(stack, `codeArtifactRepositoriesHostedZone`, {
  hostedZoneName: codeArtifactRepositoriesHostedZoneName,
  vpcId: 'Test',
  tags: [{ key: 'key', value: 'value' }],
});

new HostedZone(stack, `s3HostedHostedZone`, {
  hostedZoneName: s3HostedHostedZoneName,
  vpcId: 'Test',
  tags: [{ key: 'key', value: 'value' }],
});

new HostedZone(stack, `EcrApiHostedZone`, {
  hostedZoneName: ecrApiHostedZoneName,
  vpcId: 'Test',
  tags: [{ key: 'key', value: 'value' }],
});

new HostedZone(stack, `AppstreamApiHostedZone`, {
  hostedZoneName: appstreamApiHostedZoneName,
  vpcId: 'Test',
  tags: [{ key: 'key', value: 'value' }],
});

new HostedZone(stack, `DeviceAdvisorIotHostedZone`, {
  hostedZoneName: deviceAdvisorIotHostedZoneName,
  vpcId: 'Test',
  tags: [{ key: 'key', value: 'value' }],
});

new HostedZone(stack, `PintpointSmsVoiceV2HostedZone`, {
  hostedZoneName: pintpointSmsVoiceV2HostedZoneName,
  vpcId: 'Test',
});

new HostedZone(stack, `RumDataplaneHostedZone`, {
  hostedZoneName: rumDataplaneHostedZoneName,
  vpcId: 'Test',
});

new HostedZone(stack, `EcsAgentHostedZone`, {
  hostedZoneName: ecsAgentHostedZoneName,
  vpcId: 'Test',
});

new HostedZone(stack, `EcsTelemetryHostedZone`, {
  hostedZoneName: ecsTelemetryHostedZoneName,
  vpcId: 'Test',
});

new HostedZone(stack, `SageMakerNoteBookHostedZone`, {
  hostedZoneName: sageMakerNotebookHostedZoneName,
  vpcId: 'Test',
});

new HostedZone(stack, `SageMakerStudioHostedZone`, {
  hostedZoneName: sageMakerStudioHostedZoneName,
  vpcId: 'Test',
});

/**
 * HostedZone construct test
 */
describe('HostedZone', () => {
  snapShotTest(testNamePrefix, stack);
});
