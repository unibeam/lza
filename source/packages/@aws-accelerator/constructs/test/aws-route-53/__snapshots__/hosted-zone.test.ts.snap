// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HostedZone Construct(HostedZone):  Snapshot Test 1`] = `
{
  "Resources": {
    "AppstreamApiHostedZoneEC555793": {
      "Properties": {
        "HostedZoneTags": [
          {
            "Key": "key",
            "Value": "value",
          },
        ],
        "Name": {
          "Fn::Join": [
            "",
            [
              "appstream2.",
              {
                "Ref": "AWS::Region",
              },
              ".amazonaws.com",
            ],
          ],
        },
        "VPCs": [
          {
            "VPCId": "Test",
            "VPCRegion": {
              "Ref": "AWS::Region",
            },
          },
        ],
      },
      "Type": "AWS::Route53::HostedZone",
    },
    "DeviceAdvisorIotHostedZoneC6F4CC0C": {
      "Properties": {
        "HostedZoneTags": [
          {
            "Key": "key",
            "Value": "value",
          },
        ],
        "Name": {
          "Fn::Join": [
            "",
            [
              "deviceadvisor.iot.",
              {
                "Ref": "AWS::Region",
              },
              ".amazonaws.com",
            ],
          ],
        },
        "VPCs": [
          {
            "VPCId": "Test",
            "VPCRegion": {
              "Ref": "AWS::Region",
            },
          },
        ],
      },
      "Type": "AWS::Route53::HostedZone",
    },
    "EcrApiHostedZoneD58C89CF": {
      "Properties": {
        "HostedZoneTags": [
          {
            "Key": "key",
            "Value": "value",
          },
        ],
        "Name": {
          "Fn::Join": [
            "",
            [
              "api.ecr.",
              {
                "Ref": "AWS::Region",
              },
              ".amazonaws.com.",
            ],
          ],
        },
        "VPCs": [
          {
            "VPCId": "Test",
            "VPCRegion": {
              "Ref": "AWS::Region",
            },
          },
        ],
      },
      "Type": "AWS::Route53::HostedZone",
    },
    "EcsAgentHostedZone078A81E8": {
      "Properties": {
        "Name": {
          "Fn::Join": [
            "",
            [
              "ecs-a.",
              {
                "Ref": "AWS::Region",
              },
              ".amazonaws.com",
            ],
          ],
        },
        "VPCs": [
          {
            "VPCId": "Test",
            "VPCRegion": {
              "Ref": "AWS::Region",
            },
          },
        ],
      },
      "Type": "AWS::Route53::HostedZone",
    },
    "EcsTelemetryHostedZone654D1D44": {
      "Properties": {
        "Name": {
          "Fn::Join": [
            "",
            [
              "ecs-t.",
              {
                "Ref": "AWS::Region",
              },
              ".amazonaws.com",
            ],
          ],
        },
        "VPCs": [
          {
            "VPCId": "Test",
            "VPCRegion": {
              "Ref": "AWS::Region",
            },
          },
        ],
      },
      "Type": "AWS::Route53::HostedZone",
    },
    "PintpointSmsVoiceV2HostedZone9DE1AD33": {
      "Properties": {
        "Name": {
          "Fn::Join": [
            "",
            [
              "sms-voice.",
              {
                "Ref": "AWS::Region",
              },
              ".amazonaws.com",
            ],
          ],
        },
        "VPCs": [
          {
            "VPCId": "Test",
            "VPCRegion": {
              "Ref": "AWS::Region",
            },
          },
        ],
      },
      "Type": "AWS::Route53::HostedZone",
    },
    "RumDataplaneHostedZone1346DE6C": {
      "Properties": {
        "Name": {
          "Fn::Join": [
            "",
            [
              "dataplane.rum.",
              {
                "Ref": "AWS::Region",
              },
              ".amazonaws.com",
            ],
          ],
        },
        "VPCs": [
          {
            "VPCId": "Test",
            "VPCRegion": {
              "Ref": "AWS::Region",
            },
          },
        ],
      },
      "Type": "AWS::Route53::HostedZone",
    },
    "SageMakerNoteBookHostedZone355C458F": {
      "Properties": {
        "Name": {
          "Fn::Join": [
            "",
            [
              "notebook.",
              {
                "Ref": "AWS::Region",
              },
              ".sagemaker.aws",
            ],
          ],
        },
        "VPCs": [
          {
            "VPCId": "Test",
            "VPCRegion": {
              "Ref": "AWS::Region",
            },
          },
        ],
      },
      "Type": "AWS::Route53::HostedZone",
    },
    "SageMakerStudioHostedZone98963296": {
      "Properties": {
        "Name": {
          "Fn::Join": [
            "",
            [
              "studio.",
              {
                "Ref": "AWS::Region",
              },
              ".sagemaker.aws",
            ],
          ],
        },
        "VPCs": [
          {
            "VPCId": "Test",
            "VPCRegion": {
              "Ref": "AWS::Region",
            },
          },
        ],
      },
      "Type": "AWS::Route53::HostedZone",
    },
    "TestHostedZone68F306E4": {
      "Properties": {
        "Name": "s3-global.accesspoint.amazonaws.com",
        "VPCs": [
          {
            "VPCId": "Test",
            "VPCRegion": {
              "Ref": "AWS::Region",
            },
          },
        ],
      },
      "Type": "AWS::Route53::HostedZone",
    },
    "codeArtifactApiHostedZoneD0D526E8": {
      "Properties": {
        "HostedZoneTags": [
          {
            "Key": "key",
            "Value": "value",
          },
        ],
        "Name": {
          "Fn::Join": [
            "",
            [
              "codeartifact.",
              {
                "Ref": "AWS::Region",
              },
              ".amazonaws.com",
            ],
          ],
        },
        "VPCs": [
          {
            "VPCId": "Test",
            "VPCRegion": {
              "Ref": "AWS::Region",
            },
          },
        ],
      },
      "Type": "AWS::Route53::HostedZone",
    },
    "codeArtifactRepositoriesHostedZoneF088B188": {
      "Properties": {
        "HostedZoneTags": [
          {
            "Key": "key",
            "Value": "value",
          },
        ],
        "Name": {
          "Fn::Join": [
            "",
            [
              "d.codeartifact.",
              {
                "Ref": "AWS::Region",
              },
              ".amazonaws.com",
            ],
          ],
        },
        "VPCs": [
          {
            "VPCId": "Test",
            "VPCRegion": {
              "Ref": "AWS::Region",
            },
          },
        ],
      },
      "Type": "AWS::Route53::HostedZone",
    },
    "s3HostedHostedZoneC29BF087": {
      "Properties": {
        "HostedZoneTags": [
          {
            "Key": "key",
            "Value": "value",
          },
        ],
        "Name": {
          "Fn::Join": [
            "",
            [
              "s3.",
              {
                "Ref": "AWS::Region",
              },
              ".amazonaws.com",
            ],
          ],
        },
        "VPCs": [
          {
            "VPCId": "Test",
            "VPCRegion": {
              "Ref": "AWS::Region",
            },
          },
        ],
      },
      "Type": "AWS::Route53::HostedZone",
    },
  },
}
`;
